Configurations/unix-Makefile.tmpl

```diff
@@ -188,61 +188,111 @@ HTMLSUFFIX=html
-CPPFLAGS_Q={- $cppflags =~ s|([\\"])|\\$1|g; $cppflags -}
-CC= $(CROSS_COMPILE){- $config{cc} -}
-CFLAGS={- join(' ', @{$config{cflags}}) -}
-CXX={- $config{cxx} ? "\$(CROSS_COMPILE)$config{cxx}" : '' -}
-CXXFLAGS={- join(' ', @{$config{cxxflags}}) -}
-LDFLAGS= {- join(' ', @{$config{lflags}}) -}
-PLIB_LDFLAGS= {- join(' ', @{$config{plib_lflags}}) -}
-EX_LIBS= {- join(' ', @{$config{ex_libs}}) -}
-
-LIB_CPPFLAGS={- join(' ', '$(CPPFLAGS)',
-                          $target{shared_cppflag} || (),
-                          (map { '-D'.$_ }
-                               ('OPENSSLDIR="\"$(OPENSSLDIR)\""',
-                                'ENGINESDIR="\"$(ENGINESDIR)\""'))) -}
-LIB_CFLAGS={- join(' ', '$(CFLAGS)', $target{shared_cflag} || ()) -}
-LIB_CXXFLAGS={- join(' ', '$(CXXFLAGS)', $target{shared_cxxflag} || ()) -}
-LIB_LDFLAGS={- join(' ', '$(LDFLAGS)', $target{shared_ldflag} || (), $config{shared_ldflag} || ()) -}
-DSO_CPPFLAGS={- join(' ', '$(CPPFLAGS)', $target{dso_cppflags} || ()) -}
-DSO_CFLAGS={- join(' ', '$(CFLAGS)', $target{dso_cflags} || ()) -}
-DSO_CXXFLAGS={- join(' ', '$(CXXFLAGS)', $target{dso_cxxflags} || ()) -}
-DSO_LDFLAGS={- join(' ', '$(LDFLAGS)', $target{dso_lflags} || ()) -}
-BIN_CPPFLAGS={- join(' ', '$(CPPFLAGS)', $target{bin_cppflags} || ()) -}
-BIN_CFLAGS={- join(' ', '$(CFLAGS)', $target{bin_cflags} || ()) -}
-BIN_CXXFLAGS={- join(' ', '$(CXXFLAGS)', $target{bin_cxxflags} || ()) -}
-BIN_LDFLAGS={- join(' ', '$(LDFLAGS)', $target{bin_lflags} || ()) -}
+                                 (map { "-D".$_} @{$config{CPPDEFINES}}),
+                                 (map { "-I".$_} @{$config{CPPINCLUDES}}),
+                                 @{$config{CPPFLAGS}}) -}
+CFLAGS={- join(' ', @{$config{CFLAGS}}) -}
+CXXFLAGS={- join(' ', @{$config{CXXFLAGS}}) -}
+LDFLAGS= {- join(' ', @{$config{LDFLAGS}}) -}
+EX_LIBS= {- join(' ', @{$config{LDLIBS}}) -}
+
+MAKEDEPEND={- $config{makedepprog} -}
 
 PERL={- $config{perl} -}
 
-AR=$(CROSS_COMPILE){- $config{ar} -}
-ARFLAGS= {- join(' ', @{$config{arflags}}) -}
-RANLIB={- $config{ranlib} ? "\$(CROSS_COMPILE)$config{ranlib}" : "true"; -}
-RC= $(CROSS_COMPILE){- $target{rc} || "windres" -}
-RCFLAGS={- join(' ', @{$config{rcflags}}) -} {- $target{shared_rcflag} -}
+AR=$(CROSS_COMPILE){- $config{AR} -}
+ARFLAGS= {- join(' ', @{$config{ARFLAGS}}) -}
+RANLIB={- $config{RANLIB} ? "\$(CROSS_COMPILE)$config{RANLIB}" : "true"; -}
+RC= $(CROSS_COMPILE){- $config{RC} -}
+RCFLAGS={- join(' ', @{$config{RCFLAGS}}) -} {- $target{shared_rcflag} -}
+
+# We let the C compiler driver to take care of .s files. This is done in
+# order to be excused from maintaining a separate set of architecture
+# dependent assembler flags. E.g. if you throw -mcpu=ultrasparc at SPARC
+# gcc, then the driver will automatically translate it to -xarch=v8plus
+# and pass it down to assembler.
+AS={- $config{AS} ? "\$(CROSS_COMPILE)$config{AS}" : '$(CC) -c' -}
+ASFLAGS={- join(' ', @{$config{ASFLAGS}}) || '$(CFLAGS)' -}
```

Comment:
This can't be right. Or more specifically what can't be right is to pass overridable CFLAGS alone to $(CC). One might be able to argue in favour of passing overridable ASFLAGS to overridable $(AS), but not to $(CC). The reason is given in commentary above, there ought to be non-overridable flags present. Well, $(AS) $(ASFLAGS) doesn't seem to be used, at least not on couple of platforms I tested, x86_64 and aarch64. So question might as well be about removal of this, as opposite to fixing it that is.

Reply:
It's true that we don't use `$(AS)` or `$(ASFLAGS)` in unix-Makefile.tmpl, we rely on `$(CC)` to do the right thing.  So yeah, I agree that `AS` and `ASFLAGS` can be removed entirely.

Reply:
...  what I wonder, thought, is if that's the right thing to do.  Wouldn't *the right thing* be to use `$(AS)` and `$(whatever_ASFLAGS)` when compiling assembler files?  Considering it defaults to `$(CC)` and `$(CFLAGS) `, it will not make a practical difference unless someone actively changes AS and ASFLAGS.

Reply:
> Wouldn't the right thing be to use $(AS) and $(whatever_ASFLAGS) when compiling assembler files? Considering it defaults to $(CC) and $(CFLAGS),

As already said, one *can* argue in favour of passing overridable flag to overridable tool (so that user bear full responsibility), but it may not default to overridable $(CC) $(CFLAGS) combo. Because we need things like -march, -m64 and alike. Then you mention $(whatever_ASFLAGS), which kind of asks for non-overridable assembler flags. So is suggestion that we would have to maintain non-overridable assember flags for cases when user chooses to override AS and ASFLAGS? [Keep in mind that assemblers tend to have own flags, so you can't just copy-paste few flags from cflags.] In other words *not* using $(AS) is adequate solution, it serves us well so far, while switching to $(AS) arguably raises more questions that we are willing to answer.

Reply:
> ... it served us well so far ...

That's enough of an argument for me, and considering we have *never* used them on Unix (sure, there are ASFLAGS assignments all over the Makefiles, but no use of it anywhere...).